utils/src/activityLog/activities/ExecuteActivity.ts

```diff
@@ -0,0 +1,80 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { TreeItemCollapsibleState } from "vscode";
+import * as types from '../../../index';
+import { AppResource } from "../../../unified";
+import { AzExtParentTreeItem } from "../../tree/AzExtParentTreeItem";
+import { GenericTreeItem } from "../../tree/GenericTreeItem";
+import { nonNullProp } from "../../utils/nonNull";
+import { ActivityBase, ActivityTreeItemOptions } from "../Activity";
+
+
+interface ExecuteActivityData<C extends types.IActionContext> {
+    title: string;
+    context: C;
+}
+
+export class ExecuteActivity<C extends types.IActionContext> extends ActivityBase {
+
+    public constructor(private readonly data: ExecuteActivityData<C>, task: types.ActivityTask) {
+        super(task);
+    }
+
+    public initialState(): ActivityTreeItemOptions {
+        return {
+            label: this.data.title,
+            collapsibleState: TreeItemCollapsibleState.None,
+        }
+    }
+
+    public successState(): ActivityTreeItemOptions {
+        return {
+            label: this.labelOnDone,
+            collapsibleState: this.data.context['activityResult'] ? TreeItemCollapsibleState.Expanded : TreeItemCollapsibleState.None,
+            children: (parent: AzExtParentTreeItem) => {
+                if (this.data.context['activityResult']) {
+                    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+                    const result = this.data.context['activityResult'];
+                    const appResource: AppResource = {
+                        id: nonNullProp(result, 'id') as string,
+                        name: nonNullProp(result, 'name') as string,
+                        type: nonNullProp(result, 'type') as string,
+                    }
+
+                    const ti = new GenericTreeItem(parent, {
+                        contextValue: 'executeResult',
+                        label: "Click to view resource",
```

Comment:
I think this should be a flag that the user passes in.  It wouldn't always make sense to have this, like for a delete as an example.

Don't forget to localize!